pkg/api/grafana_com_proxy.go

```diff
@@ -24,8 +29,101 @@ var grafanaComProxyTransport = &http.Transport{
 	TLSHandshakeTimeout: 10 * time.Second,
 }
 
-func ReverseProxyGnetReq(logger log.Logger, proxyPath string, version string) *httputil.ReverseProxy {
-	url, _ := url.Parse(setting.GrafanaComUrl)
+func (hs *HTTPServer) ListGnetPlugins(c *models.ReqContext) {
+	proxy := hs.ReverseProxyGnetReq(c.Logger, "/plugins")
```

Comment:
This feels a bit awkward. Did you consider using a HTTP client rather than a reverse proxy, e.g.

```go
	c := http.Client{Transport: grafanaComProxyTransport}
	req := http.NewRequestWithContext(ctx, http.MethodGet, hs.Cfg.GrafanaComURL + "/api/plugins?<query parameters)", nil)
	resp, err := c.Do(req)
```